src/lib/images/proxy.mjs

```diff
@@ -0,0 +1,115 @@
+import express from 'express'
+import { createIPX, ipxFSStorage, ipxHttpStorage, createIPXNodeServer } from 'ipx'
+
+import { log, NETLIFYDEVERR } from '../../utils/command-helpers.mjs'
+
+export const parseAllDomains = function (config) {
+  const domains = config?.images?.remote_images
+  if (!domains) {
+    return { errors: [], remoteDomains: [] }
+  }
+
+  const remoteDomains = []
+  const errors = []
+
+  for (const patternString of domains) {
+    try {
+      const url = new URL(patternString)
+      if (url.hostname) {
+        remoteDomains.push(url.hostname)
+      } else {
+        errors.push(`The URL '${patternString}' does not have a valid hostname.`)
+      }
+    } catch (error) {
+      errors.push(`Invalid URL '${patternString}': ${error.message}`)
+    }
+  }
+
+  return { errors, remoteDomains }
+}
+
+const getErrorMessage = function ({ message }) {
+  return message
+}
+
+export const handleImageDomainsErrors = async function (errors) {
+  if (errors.length === 0) {
+    return
+  }
+
+  const errorMessage = await errors.map(getErrorMessage).join('\n\n')
+  log(NETLIFYDEVERR, `Image domains syntax errors:\n${errorMessage}`)
+}
+
+export const parseRemoteImageDomains = async function ({ config }) {
+  if (!config) {
+    return []
+  }
+
+  const { errors, remoteDomains } = await parseAllDomains(config)
+  await handleImageDomainsErrors(errors)
+
+  return remoteDomains
+}
+
+const imageUrlPattern = '/.netlify/images'
```

Comment:
[dust] I'd make this an all caps const (`export const IMAGE_URL_PATTERN = '/.netlify/images'`) at the top of the file as we have this as a convention in the repo - easier to find if we need to extend this and use it across files